anomalydetection/hampel.py

```diff
@@ -58,3 +59,39 @@ def validate_arguments(window_size, threshold):
     else:
         if threshold < 0:
             raise ValueError("Threshold must be positive.")
+
+
+@jit(nopython=True)
+def detect_using_numba(time_series, window_size, threshold=3, k=GAUSSIAN_SCALE_FACTOR):
+    """
+    Hampel filter implementation that works on numpy arrays, implemented with numba. Snatched from this implementation:
+    https://github.com/erykml/medium_articles/blob/master/Machine%20Learning/outlier_detection_hampel_filter.ipynb
+
+    Parameters
+    ----------
+    time_series: numpy.ndarray
+    window_size: int
+        The window range is from [(i - window_size):(i + window_size)], so window_size is half of the
+        window, counted in number of array elements (as opposed to specify a time span, which is not
+        supported by this implementation)
+    threshold: float
+        The threshold for marking an outlier. A low threshold "narrows" the band within which values are deemed as
+        outliers. n_sigmas
+    k
+    """
+
+    time_series_clean = time_series.copy()
+    outlier_indices = []
+    is_outlier = [False] * len(time_series)
+
+    for t in range(window_size, (len(time_series) - window_size)):
+        time_series_window = time_series[(t - window_size):(t + window_size)]
+        median_in_window = np.nanmedian(time_series_window)
```

Comment:
the non-numba implementation uses np.median, the numba implementation uses np.nanmedian. In order to align the two implementations, shouldn't we settle for one of them? Any arguments for which one?

Reply:
Do we want to allow nans as input? how do we classify nan data points, anomaly or not? or maybe nan, so that nan data points also get classified as nan anomalies? In that case I would vote for nanmedian, in order to be more robust.

Reply:
I think we should accept nans in this algorithm to make it robust. There should be a separat nan-detector then.

Reply:
@akfDHI Do you have more changes to this PR in your fork? Otherwise I will change to np.nanmedian and close this.

Reply:
@rhaDHI I have no fork and no further changes